### PR TITLE
Remove usage of 52lts compatibility layer by Avocado-VT itself [v2]

### PR DESCRIPTION
**Pull request by clebergnu**
Because Avocado 52lts has not been supported for a while, we should
drop the compatibility layer.  This is a first stage, which removes
the *usage* of that layer by Avocado itself.

The test providers also need to drop the usage of the 52lts
compatibility functions, and then we can remove it from Avocado-VT
itself.

Even the staging libraries were addressed, because it makes it allows
the compatibility module to be removed, and thus, prove that there are
no leftovers, and that there won't be any regressions.  The staging
libraries situation still need to be addressed in the sense that they
need to be synced or ideally moved to avocado.utils.

This still deserves some more testing, and that is really appreciated,
but I'm much more confident with it.  I've checked for obvious errors,
common on this type of mass change, with:

  inspekt lint --disable W,R,C,E1002,E1101,E1103,E1120,F0401,I0011,E1003

And there are no new issues introduced with this change.  I've also
checked and tp-qemu and tp-libvirt are already free of compat_52lts.

Signed-off-by: Cleber Rosa <crosa@redhat.com>

---

Changes from v1 (#2450):
* Included staging libraries
* Many missing places were addressed
* Removed `compat_52lts` module for good
* A lot more testing ;)

### CODE REVIEW

**Comment by clebergnu** [comment]
@luckyh @sathnaga @chunfuwen @kylazhang @ldoktor your review and test here is much appreciated.

**Comment by chunfuwen** [comment]
@clebergnu ,thank you for bringing up those changes, dependency on 52lts in tp-libvirt has been removed.

**Comment by clebergnu** [comment]
Thanks @luckyh, much appreciated! There's a lot of further compatibility patches coming, so I'm merging this to ease things on future PRs.

File: virttest/ceph.py [hunk 1]
```diff
@@ -167,7 +166,7 @@ def rbd_image_map(ceph_monitor, rbd_pool_name, rbd_image_name):
     """
     cmd = "rbd map %s --pool %s -m %s" % (rbd_image_name, rbd_pool_name,
                                           ceph_monitor)
-    output = decode_to_text(process.system_output(cmd, verbose=True))
+    output = process.system_output(cmd, verbose=True).stdout_text
```
Comment (YongxueHong) [inline virttest/ceph.py:169]: Hi, @clebergnu 
I saw the return of `process.system_output` is `cmd_result.stdout,` so I wonder whether `CmdResult.stdout` has the attribute `stdout_text`, from codes, seems don't have it.
my testing:
```
Python 3.6.8 (default, Dec  5 2019, 15:45:45) 
[GCC 8.3.1 20191121 (Red Hat 8.3.1-5)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from avocado.utils import process
>>> process.system_output('uname -r', verbose=True).stdout_text
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'bytes' object has no attribute 'stdout_text'
>>> 
```

Thanks.

Comment (clebergnu) [reply virttest/ceph.py:169]: @YongxueHong this is certainly a mistake, it should be `process.run().stdout_text`.  I'll send a PR fixing that.

**Pull request merged**